core/src/test/scala/com/softwaremill/events/EventMachineTest.scala

```diff
@@ -90,6 +93,36 @@ class EventMachineTest extends FlatSpec with Matchers with SqlSpec with Eventual
       m.asyncEventRunner.stop()
     }
   }
+
+  it should "recover stored events" in {
+    // given
+    var actions = Vector.empty[String]
+
+    val m = createModules(Registry()
+      .registerModelUpdate[Event1] { e => DBIOAction.successful(()).map { r => actions :+= "mu1"; r } }
+      .registerEventListener[Event1] { e => DBIOAction.successful(Nil).map { r => actions :+= "el1"; r } })
+
+    implicit val hc = HandleContext.System
+    val result = m.eventMachine.run(CommandResult.successful((), Event(Event1("x")).forNewAggregate)).futureValue
+    val result1 = m.eventMachine.run(CommandResult.successful((), Event(Event1("y")).forNewAggregate)).futureValue
+
+    val stack = new mutable.Stack[Try[Unit]]
+    val pf: PartialFunction[Try[Unit], Unit] = { case dbResponse => stack.push(dbResponse) }
+
+    // when
+    val failOverFuture = m.eventMachine.failOverFromStoredEventsTo(logResult = pf)
```

Comment:
I'm not sure if using a "logging backdoor" is a good idea to test this hear. Why not just check `actions`, which is currently never read?